railties/lib/rails/generators/rails/app/templates/Gemfile

```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
-ruby <%= "'#{RUBY_VERSION}'" %>
+ruby <%= "'#{RUBY_VERSION}'" -%>
```

Comment:
While this works for `bundle exec rails new destination`, since that includes a `gem.comment` (`# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'`), if you do `bundle exec rails new destination --dev` with your change, there's no empty line at all after `ruby '2.4.1'`.  Can you instead remove the empty line following `<% if gem.comment -%>`? It then gives a single blank line in both cases.